### PR TITLE
Either disable default mappings or disable fancy cmds

### PR DESCRIPTION
**Pull request by juanMarinero**
This requires an specific codeline order in ~/.vimrc:
RIGHT order:
    let g:loaded_painter_default_mappings = 0
    Plug 'lpinilla/vim-codepainter'
WRONG order:
    Plug 'lpinilla/vim-codepainter'
    let g:loaded_painter_default_mappings = 0
I'm newbie with Vim-plugins, this order dependent actually should be fixed.

### CODE REVIEW

**Comment by lpinilla** [comment]
I like the approach, I agree that the order shouldn't determine the behavior.

The only thing I would change is to replace the word fancy for "default", so it would be default mappings and default commands.

**Comment by juanMarinero** [comment]
I was wrong! The `Plug ` commands in `~/.vimrc`  (like `Plug 'lpinilla/vim-codepainter'`) are loaded **after** reading (most part) of that file. I'm sorry, I do not remember where just few days before I have read it.

TLDR, this is fine:
```
Plug 'lpinilla/vim-codepainter'
" unset to disable default mappings
let g:loaded_painter_default_mappings = 0
```

**Comment by juanMarinero** [comment]
Btw I think this is a good place to share my customized mappings

```
" {mm} to be consistent with vim-highlightedyank plugin
                                                                                                                                                                                                                                                                                                                                                                                                                                                          
" highlight visual selection
vnoremap <silent> <leader>mm :<c-u> call codepainter#paintText(visualmode())<cr>
" toogle to see highlight text
nnoremap <silent> <leader>mmt :<c-u> call codepainter#paintText('')<cr>
" go to next highlighted in file
nnoremap <silent> <leader>mmg :<c-u> call codepainter#navigate()<cr>
" erase highlights in current line
nnoremap <silent> <leader>mmdl :<c-u>PainterEraseLine<cr>
" erase highlights in whole file
nnoremap <silent> <leader>mmda! :<c-u>PainterEraseAll<cr>
                                                                                                                                                                                                                                                                                                                                                                                                                                                          
" pick another highlight color
nnoremap <silent> <leader>mm1 :<c-u> PainterPickColor<space>1 <cr>
nnoremap <silent> <leader>mm2 :<c-u> PainterPickColor<space>2 <cr>
nnoremap <silent> <leader>mm3 :<c-u> PainterPickColor<space>3 <cr>
nnoremap <silent> <leader>mm4 :<c-u> PainterPickColor<space>4 <cr>
nnoremap <silent> <leader>mm5 :<c-u> PainterPickColor<space>5 <cr>
nnoremap <silent> <leader>mm6 :<c-u> PainterPickColor<space>6 <cr>
nnoremap <silent> <leader>mm7 :<c-u> PainterPickColor<space>7 <cr>
nnoremap <silent> <leader>mm8 :<c-u> PainterPickColor<space>8 <cr>
nnoremap <silent> <leader>mm9 :<c-u> PainterPickColor<space>9 <cr>
nnoremap <silent> <leader>mm10 :<c-u> PainterPickColor<space>10 <cr>
                                                                                                                                                                                                                                                                                                                                                                                                                                                          
" load marks from JSON
nnoremap <silent> <leader>mml :<c-u> PainterLoadMarks<space>_vim-codepainter.json <bar> echo "file of marks to load"<Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left>

" save marks to JSON
nnoremap <silent> <leader>mms :<c-u> PainterSaveMarks<space>_vim-codepainter.json <bar> echo "save marks to this file"<Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left><Left>
```

**Comment by juanMarinero** [comment]
> replace the word fancy for "default"

Ehhh, actually it was a dirty merge proposal, I didn't expect you fancy it hehe.

There are several further to check (note I'm newbie about vim plug coding):
- initial lines talk about `script`, but actually one `if` checks cmds, and other checks mappings. Also, maybe is better to make it more modular and split whole file into two (with a more representative name than `codepainter_1.vim` and `codepainter_2.vim`)
```
" check whether this script is already loaded
" or the user doesn't want to load it
```
- is `cmd` acceptable? I though it was like an slang (English is not my mother language)
```
" fancy cmds {{{1
if !exists("g:loaded_painter_fancy_cmds")
  let g:loaded_painter_fancy_cmds = 1
```

- man use `maps` or `mappings`? 
```
" default mappings {{{1
if !exists("g:loaded_painter_default_mappings")
  let g:loaded_painter_default_mappings = 1
```

Summary, I'm glad you like my proposal, but I think is better that you (a non-newbie and good English speaker) edit the file/s. Thanks!

**Comment by lpinilla** [comment]
I like the approach but I think that having two files would be an overhead for simple defines, if there is no issue with loading the entire file, I prefer to keep it that way to make it more simple to follow.

Thanks for your contributions but this time I'm choosing to keep it simple. Good work anyways!

**Comment by juanMarinero** [comment]
Previous load and save maps that  I shared are **NOT** right, since the respective plugin functions accept undefined number of parameters, and the `| echo` is understood as a parameter, see: [codepainter.vim codepainter#SaveMarks(...)](https://github.com/lpinilla/vim-codepainter/blob/master/autoload/codepainter.vim#L195) function declaration  (same for [codepainter#LoadMarks(...)](https://github.com/lpinilla/vim-codepainter/blob/master/autoload/codepainter.vim#L210))

TLDR
Wrong: `nnoremap <silent> <leader>mml :<c-u>PainterLoadMarks<space>vim-codepainter.json <bar> echo "blabla"<cr>`
Right:   `nnoremap <silent> <leader>mml :<c-u>PainterLoadMarks<space>vim-codepainter.json<cr>`
Wrong: `nnoremap <silent> <leader>mml :<c-u>PainterSaveMarks<space>vim-codepainter.json <bar> echo "blabla"<cr>`
Right:   `nnoremap <silent> <leader>mml :<c-u>PainterSaveMarks<space>vim-codepainter.json<cr>`

Sorry for the inconveniences!

**Comment by juanMarinero** [comment]
> I like the approach but I think that having two files would be an overhead for simple defines, if there is no issue with loading the entire file, I prefer to keep it that way to make it more simple to follow.

then, maybe as just one file, changing from my branch:
> The only thing I would change is to replace the word fancy for "default", so it would be default mappings and default commands.
